src/pymmcore_widgets/_stage_widget.py

```diff
@@ -21,37 +22,150 @@
-STAGE_DEVICES = {DeviceType.Stage, DeviceType.XYStage}
-STYLE = """
-QPushButton {
-    border: none;
-    background: transparent;
-    color: rgb(0, 180, 0);
-    font-size: 40px;
+CORE = Keyword.CoreDevice
+XY_STAGE = Keyword.CoreXYStage
+FOCUS = Keyword.CoreFocus
+
+MOVE_BUTTONS: dict[str, tuple[int, int, int, int]] = {
+    # btn glyph                (r, c, xmag, ymag)
+    MDI6.chevron_triple_up: (0, 3, 0, 3),
+    MDI6.chevron_double_up: (1, 3, 0, 2),
+    MDI6.chevron_up: (2, 3, 0, 1),
+    MDI6.chevron_down: (4, 3, 0, -1),
+    MDI6.chevron_double_down: (5, 3, 0, -2),
+    MDI6.chevron_triple_down: (6, 3, 0, -3),
+    MDI6.chevron_triple_left: (3, 0, -3, 0),
+    MDI6.chevron_double_left: (3, 1, -2, 0),
+    MDI6.chevron_left: (3, 2, -1, 0),
+    MDI6.chevron_right: (3, 4, 1, 0),
+    MDI6.chevron_double_right: (3, 5, 2, 0),
+    MDI6.chevron_triple_right: (3, 6, 3, 0),
 }
-QPushButton:hover:!pressed {
-    color: rgb(0, 255, 0);
-}
-QPushButton:pressed {
-    color: rgb(0, 100, 0);
-}
-QSpinBox {
-    min-width: 35px;
-    height: 22px;
-}
-QLabel {
-    color: #999;
-}
-QCheckBox {
-    color: #999;
-}
-QCheckBox::indicator {
-    width: 11px;
-    height: 11px;
-}
-"""
+
+
+class MoveStageButton(QPushButton):
+    def __init__(self, glyph: str, xmag: int, ymag: int, parent: QWidget | None = None):
+        super().__init__(parent=parent)
+        self.xmag = xmag
+        self.ymag = ymag
+        self.setAutoRepeat(True)
+        self.setFlat(True)
+        self.setFocusPolicy(Qt.FocusPolicy.NoFocus)
+        self.setCursor(Qt.CursorShape.PointingHandCursor)
+        setTextIcon(self, glyph)
+        self.setStyleSheet(
+            """
+            MoveStageButton {
+                border: none;
+                background: transparent;
+                color: rgb(0, 180, 0);
+                font-size: 40px;
+            }
+            MoveStageButton:hover:!pressed {
+                color: rgb(0, 255, 0);
+            }
+            MoveStageButton:pressed {
+                color: rgb(0, 100, 0);
+            }
+            """
+        )
+
+
+class StageMovementButtons(QWidget):
+    """Grid of buttons to move a stage in 2D.
+
+            ^
+    << < [dstep] > >>
+            v
+    """
+
+    moveRequested = Signal(float, float)
+
+    def __init__(
+        self, levels: int = 2, show_x: bool = True, parent: QWidget | None = None
+    ) -> None:
+        super().__init__(parent)
+        self._levels = levels
+        self._x_visible = show_x
+
+        btn_grid = QGridLayout(self)
+        btn_grid.setContentsMargins(0, 0, 0, 0)
+        btn_grid.setSpacing(0)
+        for glyph, (row, col, xmag, ymag) in MOVE_BUTTONS.items():
+            btn = MoveStageButton(glyph, xmag, ymag)
+            btn.clicked.connect(self._on_move_btn_clicked)
+            btn_grid.addWidget(btn, row, col, Qt.AlignmentFlag.AlignCenter)
+
+        # step size spinbox in the middle of the move buttons
+        self.step_size = QDoubleSpinBox()
```

Comment:
> add a unit here

👍 

> Is it possible to add some sort of indication (spin buttons?) to indicate that this value can be edited

yeah, it's possible.  again, i rather dislike the look/UX of spin buttons; and I do think that the fact that the cursor turns into the little text cursor icon when you hover over it goes a long way towards making you realize that.  One is very likely to accidentally brush over the field while using the widget, and then they know forever that it's editable.  so, would love to avoid adding the spin buttons if at all possible.  if the hover effects aren't enough, let's brainstorm on other ways to make it obvious without spin buttons

Reply:
btw, can't show it in a screenshot, but this is what the cursor looks like when you hover past that field:

![cursor-ibeam](https://github.com/user-attachments/assets/1945b948-27f3-4b96-9242-6211e22a5e2a)